src/platform/posix/tcp_ip_channel.c

```diff
@@ -21,29 +21,25 @@
 
 #define MIN(a, b) ((a) < (b) ? (a) : (b))
 
+static bool _is_globals_initialized = false;
+static Environment *_env;
+
 // Forward declarations
-static void TcpIpChannel_spawn_receive_thread(TcpIpChannel *self);
-static lf_ret_t TcpIpChannel_reset_socket(TcpIpChannel *self);
-static void *TcpIpChannel_receive_thread(void *untyped_self);
-
-static void TcpIpChannel_socket_set_blocking(int fd, bool blocking) {
-  // Set socket to blocking
-  int opts = fcntl(fd, F_GETFL);
-  if (opts < 0) {
-    throw("Could not get socket options");
-  }
-  if (blocking) {
-    opts = (opts & (~O_NONBLOCK));
-  } else {
-    opts = (opts | O_NONBLOCK);
-  }
```

Comment:
This is not needed anymore since we can just always block inside of the TcpIpChannel now. It also makes handling the state more easy. The runtime is not blocked by it anymore.